reana_workflow_controller/rest.py

```diff
@@ -927,3 +934,178 @@ def get_workflow_status(workflow_id):  # noqa
+              }
+        404:
+          description: >-
+            Request failed. Either User or Workflow doesn't exist.
+          examples:
+            application/json:
+              {
+                "message": "User 00000000-0000-0000-0000-000000000000 doesn't
+                            exist"
+              }
+            application/json:
+              {
+                "message": "Workflow 256b25f4-4cfb-4684-b7a8-73872ef455a1
+                            doesn't exist"
+              }
+        500:
+          description: >-
+            Request failed. Internal controller error.
+    """
+
+    try:
+        organization = request.args['organization']
+        user_uuid = request.args['user']
+        workflow = Workflow.query.filter(Workflow.id_ == workflow_id).first()
+        status = request.json
+        if not (status in STATUSES):
+            return jsonify({'message': 'Status {0} is not one of: {1}'.
+                            format(status, ", ".join(STATUSES))}), 400
+        if not workflow:
+            return jsonify({'message': 'Workflow {} does not exist'.
+                            format(workflow_id)}), 404
+        if not str(workflow.owner_id) == user_uuid:
+            return jsonify(
+                {'message': 'User {} is not allowed to access workflow {}'
+                 .format(user_uuid, workflow_id)}), 403
+        if status == START:
+            return start_workflow(organization, user_uuid, workflow)
+        elif status == STOP:
+            return stop_workflow(organization, user_uuid, workflow)
+        else:
+            raise NotImplemented("Status {} is not supported yet"
+                                 .format(status))
+    except KeyError as e:
+        return jsonify({"message": str(e)}), 400
+    except Exception as e:
+        return jsonify({"message": str(e)}), 500
+
+
+def start_workflow(organization, user_uuid, workflow):
+    """Start a workflow."""
+    workflow.status = WorkflowStatus.running
+    db.session.commit()
+    if workflow.type_ == 'yadage':
+        return run_yadage_workflow_from_spec(organization,
+                                             user_uuid,
+                                             workflow)
+    elif workflow.type_ == 'cwl':
+        pass
+
+
+def stop_workflow(organization, user_uuid, workflow):
+    """Stop a workflow."""
+    workflow.status = WorkflowStatus.finished
+    db.session.commit()
+    return jsonify({'message': 'Workflow stopped',
+                    'workflow_id': workflow.id_,
+                    'status': workflow.status.name,
+                    'organization': organization,
+                    'user': user_uuid}), 200
+
+
+def run_yadage_workflow_from_spec(organization, user_uuid, workflow):
+    """Run a yadage workflow."""
+    try:
+        kwargs = {
+            "workflow_workspace": workflow.workspace_path,
+            "workflow_json": workflow.specification,
+            "parameters": workflow.parameters
```

Comment:
`workflow_uuid` is missing see [here](https://github.com/reanahub/reana-workflow-engine-yadage/blob/master/reana_workflow_engine_yadage/tasks.py#L69). This will probably break when calling celery, have you tried to do the following with all your PRs integrated:
```console
~/reana-demo-root6-roofit $ reana-client -l debug workflow create -f .reana.yaml
some-uuid
~/reana-demo-root6-roofit $ reana-client workflow start --workflow some-uuid
workflow started blah blah
```
And check if it actually works?

Reply:
Indeed, I just copied the code from `run_yadage_workflow_from_spec_endpoint`, it is fixed. However, I have troubles with spawning yadage workers both without kubernetes and inside minikube, so I couldn't test the full chain. 

(the trouble is in connecting to message broker, I have UnicodeDecodeError, would be good if you could verify running/failing engine-yadage later. I don't have such a problem with engine-cwl)